### PR TITLE
[JAVA] Fixed async sample output

### PR DESCRIPTION
**Pull request by likholat**

### CODE REVIEW

File: modules/java_api/samples/README.md [hunk 1]
```diff
@@ -12,10 +12,11 @@ Create an environment variable with Inference Engine installation path:
 export IE_PATH=/path/to/openvino/bin/intel64/Release/lib
 ```
 
-To create java library and java samples for Inference Engine add `-DENABLE_JAVA=ON` flag in cmake command while building dldt:
+To create java library and java samples for Inference Engine add `--DIE_EXTRA_MODULES` flag in cmake command while building `openvino`:
```
Comment (dkurt) [inline modules/java_api/samples/README.md:15]: `-DIE_EXTRA_MODULES`

File: modules/java_api/samples/README.md [hunk 2]
```diff
@@ -12,10 +12,11 @@ Create an environment variable with Inference Engine installation path:
...
+
 ```bash
 cd /openvino/build
-cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_JAVA=ON -DENABLE_SAMPLES=ON ..
+cmake -DCMAKE_BUILD_TYPE=Release -DIE_EXTRA_MODULES=/<openvino_contrib>/modules -DENABLE_SAMPLES=ON -DENABLE_OPENCV=OFF ..
```
Comment (dkurt) [inline modules/java_api/samples/README.md:19]: remove extra /:
```
-DIE_EXTRA_MODULES=<openvino_contrib>/modules
```

File: modules/java_api/samples/face_detection_sample_async/Main.java [hunk 3]
```diff
@@ -108,6 +108,10 @@ public void run() {
                             framesCounter++;
                             framesQueue.add(frame.clone());
                         }
+                        if (framesCounter == 0) {
+                            System.out.println("ERROR: Can't get any video frame!");
```
Comment (dkurt) [inline modules/java_api/samples/face_detection_sample_async/Main.java:112]: `System.err.println`

File: modules/java_api/samples/face_detection_sample_async/Main.java [hunk 4]
```diff
@@ -232,12 +236,13 @@ public void run() {
                     Imgproc.putText(img, label1, new Point(10, 80), 0, 0.7, color, 1);
                 }
                 HighGui.imshow("Detection", img);
+                HighGui.waitKey(1);
```
Comment (dkurt) [inline modules/java_api/samples/face_detection_sample_async/Main.java:239]: Can we stop application by button? If no, can you try
```java
if (HighGui.waitKey(1) != -1) 
    break;
```
see https://docs.opencv.org/master/d7/dfc/group__highgui.html#ga5628525ad33f52eab17feebcfba38bd7

File: modules/java_api/samples/face_detection_sample_async/Main.java [hunk 5]
```diff
@@ -232,12 +236,13 @@ public void run() {
...
             }
 
             captureThread.join();
             inferThread.join();
 
-            HighGui.waitKey(0);
+            HighGui.waitKey(1);
```
Comment (dkurt) [inline modules/java_api/samples/face_detection_sample_async/Main.java:245]: Not sure if we really need `waitKey` here

File: modules/java_api/samples/face_detection_sample_async/Main.java (hunk 4 above)
Comment (likholat) [reply modules/java_api/samples/face_detection_sample_async/Main.java:239]: It doesn't work for Java as it should. But it's ok for python...

**Comment by dkurt** [comment]
Apply fix from https://github.com/openvinotoolkit/openvino_contrib/pull/14 to pass Linux tests.

File: modules/java_api/samples/face_detection_sample_async/Main.java [hunk 6]
```diff
@@ -100,14 +100,31 @@ public static void main(String[] args) {
...
-                        VideoCapture cam = new VideoCapture();
-                        cam.open(imgsPath);
+                        int camMode;
+                        try {
+                            camMode = Integer.valueOf(imgsPath);
+                        } catch (NumberFormatException exception) {
+                            camMode = -1;
+                        }
+
+                        VideoCapture cam;
```
Comment (dkurt) [inline modules/java_api/samples/face_detection_sample_async/Main.java:110]: There is override of `.open()` with integer.
```java
VideoCapture cam = new VideoCapture();
try {
    int idx = Integer.valueOf(imgsPath);
    cam.open(idx);
} catch (NumberFormatException exception) {
    cam.open(imgsPath);
}
```

File: modules/java_api/samples/face_detection_sample_async/Main.java (hunk 4 above)
Comment (dkurt) [reply modules/java_api/samples/face_detection_sample_async/Main.java]: Well, which values are returned by `waitKey` ?

File: modules/java_api/samples/face_detection_sample_async/Main.java [hunk 7]
```diff
@@ -145,8 +155,10 @@ public void run() {
                             boolean isRunning = true;
 
                             while (captureThread.isAlive() || !framesQueue.isEmpty()) {
-                                processInferRequets(WaitMode.STATUS_ONLY);
+                                if (Thread.interrupted())
+                                    break;
```
Comment (dkurt) [inline modules/java_api/samples/face_detection_sample_async/Main.java]: Shall we check capture thread in the same way?

File: modules/java_api/samples/face_detection_sample_async/Main.java (hunk 5 above)
Comment (dkurt) [reply modules/java_api/samples/face_detection_sample_async/Main.java:254]: Still have a question why we need `waitKey` here?

**Review by dkurt — APPROVED** [review]

File: modules/java_api/samples/face_detection_sample_async/Main.java (hunk 5 above)
Comment (likholat) [reply modules/java_api/samples/face_detection_sample_async/Main.java:254]: Can't destroy window without this `waitKey`

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**